.travis.yml

```diff
@@ -3,13 +3,19 @@ compiler:
   - gcc
 before_script:
   - git submodule update --init --recursive
-  - mkdir -p build
-  - cd build
-  - cmake .. -DCMAKE_BUILD_TYPE=Debug
+  - mkdir -p build/debug
+  - mkdir -p build/release
+  - cd build/debug
+  - cmake ../.. -DCMAKE_BUILD_TYPE=Debug
+  - cd ../release
+  - cmake ../.. -DCMAKE_BUILD_TYPE=Release
```

Comment:
End `before_script` with `cd ..`.  Then start `script` with `cd ./release`.  This minimizes state sharing between tasks. Otherwise `script` will break if the `before_script` is changed to finish in a different directory.

Reply:
Agreed. Good call.